x-pack/test/case_api_no_sub_cases_integration/basic/tests/cases/comments/delete_comment.ts

```diff
@@ -0,0 +1,99 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../../common/ftr_provider_context';
+
+import { CASES_URL } from '../../../../../../plugins/cases/common/constants';
+import { postCaseReq, postCommentUserReq } from '../../../../common/lib/mock';
+import {
+  deleteAllCaseItems,
+  deleteCases,
+  deleteCasesUserActions,
+  deleteComments,
+} from '../../../../common/lib/utils';
+
+// eslint-disable-next-line import/no-default-export
+export default ({ getService }: FtrProviderContext): void => {
+  const supertest = getService('supertest');
+  const es = getService('es');
+
+  describe('delete_comment', () => {
+    afterEach(async () => {
+      await deleteCases(es);
+      await deleteComments(es);
+      await deleteCasesUserActions(es);
+    });
+
+    it('should delete a comment', async () => {
+      const { body: postedCase } = await supertest
+        .post(CASES_URL)
+        .set('kbn-xsrf', 'true')
+        .send(postCaseReq)
+        .expect(200);
+
+      const { body: patchedCase } = await supertest
+        .post(`${CASES_URL}/${postedCase.id}/comments`)
+        .set('kbn-xsrf', 'true')
+        .send(postCommentUserReq)
+        .expect(200);
+
+      const { body: comment } = await supertest
+        .delete(`${CASES_URL}/${postedCase.id}/comments/${patchedCase.comments[0].id}`)
+        .set('kbn-xsrf', 'true')
+        .expect(204)
+        .send();
+      expect(comment).to.eql({});
+    });
+
+    it('unhappy path - 404s when comment belongs to different case', async () => {
+      const { body: postedCase } = await supertest
+        .post(CASES_URL)
+        .set('kbn-xsrf', 'true')
+        .send(postCaseReq)
+        .expect(200);
+
+      const { body: patchedCase } = await supertest
+        .post(`${CASES_URL}/${postedCase.id}/comments`)
+        .set('kbn-xsrf', 'true')
+        .send(postCommentUserReq)
+        .expect(200);
+
+      const { body } = await supertest
+        .delete(`${CASES_URL}/fake-id/comments/${patchedCase.comments[0].id}`)
+        .set('kbn-xsrf', 'true')
+        .send()
+        .expect(404);
+
+      expect(body.message).to.eql(
+        `This comment ${patchedCase.comments[0].id} does not exist in fake-id).`
+      );
+    });
+
+    it('unhappy path - 404s when comment is not there', async () => {
+      await supertest
+        .delete(`${CASES_URL}/fake-id/comments/fake-id`)
+        .set('kbn-xsrf', 'true')
+        .send()
+        .expect(404);
+    });
+
+    describe('sub case comments', () => {
+      afterEach(async () => {
+        await deleteAllCaseItems(es);
+      });
+
+      it('fails to delete a comment from a sub case', async () => {
+        await supertest
+          .delete(`${CASES_URL}/invalid-id/comments/invalid-comment-id?subCaseId=a-random-id`)
+          .set('kbn-xsrf', 'true')
+          .send()
+          .expect(404);
```

Comment:
The `404` will be thrown because the case does not exist or because the subCaseId does not exist?